lib/kernel/doc/src/app.xml

```diff
@@ -148,10 +149,17 @@ ApplicationVersion = string()</code>
       <tag><c>applications</c></tag>
       <item>
         <p>All applications that must be started before this
-          application is allowed to be started. <c>systools</c> uses
-          this list to generate correct start scripts. Defaults to
-          the empty list, but notice that all applications have
-          dependencies to (at least) Kernel and STDLIB.</p>
+          application. If an application is also listed in
+          <c>optional_applications</c>, then there will be an
+          attempt to start the optional application, but failing
+          to run it won't stop this application from starting.</p>
```

Comment:
I disagree here. If the optional dep is in the code path I would expect it to fail startup if that application fails to start.